### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by newhoteng**
## OOP school library: set up associations
In this milestone I:
- Created a `Classroom` class with the following:
  - @label instance variable that is initialized in the constructor
  - Setter and getter for @label
- Created `has-many/belongs-to` relationship between Classroom and Student per project requirement
- Created a `Book` class per project requirement
- Created a `Rental` class per project requirement
- Created a many-to-many relationship between `Person` and `Book` through the `Rental` class

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
```
Comment (Zuhaib042) [inline book.rb:1]: - [ ] Kindly consider adding a method `add_rental` in this class after initialize method. In that method you will be getting the `person,date` and inside that method, you will be initializing  and returning a rental object where you will be also giving the self `book class` to make the association right. example: `Rental.new(date, self, person)`

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, person
```
Comment (Zuhaib042) [inline rental.rb]: - [ ] Kindly consider prefixing person with colon to make the syntax right like `:person`.

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require_relative 'nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, rentals
   attr_accessor :name, :age
```
Comment (Zuhaib042) [inline person.rb]: - [ ] Kindly consider adding the `rentals` in the attr_accessor and dont forget to prefixing rentals with color `:` to make the syntax right.

File: person.rb [hunk 4]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (Zuhaib042) [inline person.rb:15]: - [ ] Kindly consider adding a method add_rental in this class after initialize method. In that method you will be getting the parameters of (book,date) and inside that method, you will be initializing and returning a rental object where you will be also giving the self person class to make the association right. example: Rental.new(date, self, book)

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @newhoteng 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Professional Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @newhoteng , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Setuped associations ✔️
- Well Structured Code✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**